lib/logstash/outputs/elasticsearch.rb

```diff
@@ -368,9 +370,31 @@ def submit(actions)
   def flush(actions, teardown = false)
     begin
       submit(actions)
+    rescue Manticore::SocketException => e
+      # If we can't even connect to the server let's just print out the URL (:host is actually a URL)
+      # and let the user sort it out from there
+      @logger.error(
+        "Attempted to send a bulk request to Elasticsearch configured at '#{@current_client.client_options[:host]}',"+
+          " but Elasticsearch appears to be unreachable or down!",
+        :client_config => @current_client.client_options,
+        :error_message => e.message,
+        :error_class => e.class.name,
+        :backtrace => e.backtrace
```

Comment:
Sweet, thanks for the perspective. I'm in favor of backtrace for this.

![high five](http://static1.fjcdn.com/thumbnails/comments/Large+gif+size+is+large+_904351a6217f52a42029cc79d1bdb94c.gif)